test/Test/Data/Mutable/HashMap.hs

```diff
@@ -135,14 +158,14 @@ val = do
   let strSize = Range.singleton 3
   Gen.string strSize Gen.alpha
 
--- | Random pairs
+-- | Random pairs with no duplicate keys
 keyVals :: Gen [(Int, String)]
 keyVals = do
   size <- Gen.int $ Range.linear 0 maxSize
   let sizeGen = Range.singleton size
   keys <- Gen.list sizeGen key
   vals <- Gen.list sizeGen val
-  return $ zip keys vals
+  return $ zip (nubOrd keys) vals
```

Comment:
I think it might be good to test it using duplicate keys, because the behaviour on key collisions (on `fromList` eg.) is also significant.

Reply:
Sure -- I just did it to avoid making one test more complicated to write.